t/lib/NoFork.pm

```diff
@@ -4,7 +4,21 @@ BEGIN {
     *CORE::GLOBAL::fork = sub { die "you should not fork" };
 }
 use Config;
-tied(%Config)->{d_fork} = 0;    # blatant lie
+if ($Config::Config{d_fork}) {
```

Comment:
The cleaner solution would be to use Mock::Config of course. But this is a core module, so I went with the least effort.